ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphPage.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { Flex, FlexItem, PageSection } from '@patternfly/react-core';
+
+import useURLSearch from 'hooks/useURLSearch';
+import { fetchClustersAsArray } from 'services/ClustersService';
+import { Cluster } from 'types/cluster.proto';
+import ClusterSelect from './ClusterSelect';
+import NamespaceSelect from './NamespaceSelect';
+
+export type Namespace = {
+    metadata: {
+        id: string;
+        name: string;
+    };
+};
+
+function NetworkGraphPage() {
+    const [clusters, setClusters] = React.useState<Cluster[]>([]);
+    const { searchFilter, setSearchFilter } = useURLSearch();
+    const selectedClusterId = (searchFilter.Cluster as string) || '';
+    const selectedNamespaces = (searchFilter.Namespace as string[]) || [];
+
+    React.useEffect(() => {
+        fetchClustersAsArray()
+            .then((data) => {
+                setClusters(data as Cluster[]);
+            })
+            .catch(() => {
+                // TODO
+            });
+    }, []);
+
+    function updateSelectedClusterId(selection) {
+        const newSearchFiliter = { ...searchFilter, Cluster: selection };
```

Comment:
I think there might have been a typo here for `newSearchFilter`. Same a few lines below